src/Hal/Report/Html/Reporter.php

```diff
@@ -29,47 +39,31 @@ public function __construct(Config $config, Output $output)
         $this->output = $output;
     }
 
-
+    /**
+     * {@inheritDoc}
+     */
     public function generate(Metrics $metrics)
     {
         $logDir = $this->config->get('report-html');
         if (!$logDir) {
             return;
         }
 
-        // consolidate
         $consolidated = new Consolidated($metrics);
 
-        // history of builds
-        $today = (object)[
-            'avg' => $consolidated->getAvg(),
-            'sum' => $consolidated->getSum()
-        ];
         $files = glob($logDir . '/js/history-*.json');
-        $next = count($files) + 1;
-        $history = [];
         natsort($files);
-        foreach ($files as $filename) {
-            array_push($history, json_decode(file_get_contents($filename)));
-        }
-
-        // copy sources
-        if (!file_exists($logDir . '/js')) {
-            mkdir($logDir . '/js', 0755, true);
-        }
-        if (!file_exists($logDir . '/css')) {
-            mkdir($logDir . '/css', 0755, true);
-        }
-        if (!file_exists($logDir . '/images')) {
-            mkdir($logDir . '/images', 0755, true);
-        }
-        if (!file_exists($logDir . '/fonts')) {
-            mkdir($logDir . '/fonts', 0755, true);
+        $history = array_map(static function ($filename) {
```

Comment:
why replacing all `foreach` by `array_map` ? I'm not sure that's more readable...

Reply:
IMO I prefer `array_map` over `foreach` for its ability to set typed arguments and typed return clearly in the code, while you can not do that with the foreach. Since I'm using PHP 7.1 or above, I started to naturally write my foreach loops as array_map for that reason.

I kow that's not applicatble to this project since it must supports PHP 5.5 too, but that's not the only advantage to me. 

When you must create an array looping on other one(s), using `foreach` will oblige you to firstly initialize your array, then your `foreach` will push data inside. But if it is a long loop, it can be very hard to read and find if your array to populate is not fed more than once, for example. With the `array_map`, even if you may have several `return` in your callback, you will only return 1 thing for 1 loop-process, and you can easily know what that thing is by directly spotting the `return` instructions. Also, using `array_map` will provide you directly the array you want to build instead of doing it in 2 statements: initialization, then appending.

I keep using `foreach` when I care about the keys of the array I want to create as `array_map` does not offer me that way of doing it (actually, it does, but then, it is clearly a disguised foreach...).